gohttplib.go

```diff
@@ -23,14 +23,25 @@ import (
 	"bytes"
 	"net/http"
 	"unsafe"
+	"context"
 )
 
 var cpointers = PtrProxy()
+var srv http.Server = http.Server
 
 //export ListenAndServe
 func ListenAndServe(caddr *C.char) {
 	addr := C.GoString(caddr)
-	http.ListenAndServe(addr, nil)
+	srv.Addr = addr
+	Shutdown()
+	srv.ListenAndServe()
+}
+
+// export Shutdown
+func Shutdown() {
+	http.HandleFunc("/shutdown", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This handler exposes the shutdown over HTTP, which is not something gohttplib should be doing.

Try to set it up so you can call Shutdown() from Python or C.

Reply:
Alright, so I have re-written the shutdown function so it doesn't work like rest:

https://github.com/maxknivets/gohttplib/blob/master/gohttplib.go#L40

Then I imported the function to python:
https://github.com/maxknivets/gohttplib/blob/master/examples/python/gohttp/__init__.py#L81

And finally, added the function to the list in C file:
https://github.com/maxknivets/gohttplib/blob/master/examples/python/gohttp/build_gohttplib.py#L25